### DISCUSSION TITLE
Allow third-party packages to enqueue scripts alongside Livewire

### DISCUSSION THREAD

**Discussion by ryangjchandler** [livewire/livewire#1372] [4. General]
Currently when developing a third-party package / extension for Livewire that needs both server-side and client-side logic, the developer needs to add the scripts manually.

I think it would be pretty neat if you could hook into the `<livewire:scripts />` / `@livewireScripts` methods to add your own scripts in too, makes the developer experience a bit nicer too.

**Comment by asantibanez** [comment]
Sounds really good @ryangjchandler 

The only thing is that `view:clear` would still be needed to run on each package upgrade, right?

**Reply by ryangjchandler** [reply]
@asantibanez `view:clear` should really be run on each deployment (after running `composer install`) anyway and I doubt anybody is caching views locally.
